rtrlib/rtr/rtr.c

```diff
@@ -64,22 +159,28 @@ int rtr_init(struct rtr_socket *rtr_socket,
     if(tr != NULL)
         rtr_socket->tr_socket = tr;
 
-    if(refresh_interval > 86400 || refresh_interval < 1) {
-        return RTR_INVALID_PARAM;
-    } else {
+    int interv_retval = rtr_check_interval_range(refresh_interval, REFRESH);
+
+    if(interv_retval == INSIDE_INTERVAL_RANGE) {
```

Comment:
you could summarize the check into one like:
```
if (rtr_check_interval_range(refresh_interval, REFRESH) != INSIDE_INTERVAL_RANGE || 
    rtr_check_interval_range(refresh_interval, EXPIRATION) != INSIDE_INTERVAL_RANGE || 
    rtr_check_interval_range(refresh_interval, RETRY) != INSIDE_INTERVAL_RANGE) {
    /* optional: print some error message? */
    return RTR_INVALID_PARAM;
}
/* otherwise not error, hence set all as given */
```